manual/azure-sru-19.3.41.txt

```diff
@@ -162,44 +155,56 @@ for series in $SRU_SERIES; do
 
 	echo "Created $name: ubuntu@$IP"
 
-        ssh "${sshopts[@]}" $IP -- cloud-init status --wait --long
-        ssh "${sshopts[@]}" $IP -- dpkg-query --show cloud-init
-        ssh "${sshopts[@]}" $IP -- sudo cat /run/cloud-init/result.json
-        ssh "${sshopts[@]}" $IP -- grep Trace /var/log/cloud-init.log
-        ssh "${sshopts[@]}" $IP -- systemd-analyze
-        ssh "${sshopts[@]}" $IP -- systemd-analyze blame
-        ssh "${sshopts[@]}" $IP -- cloud-init analyze show
-        ssh "${sshopts[@]}" $IP -- cloud-init analyze blame
+        ssh $sshopts $IP -- cloud-init status --wait --long
+        ssh $sshopts $IP -- dpkg-query --show cloud-init
+        ssh $sshopts $IP -- sudo cat /run/cloud-init/result.json
+        ssh $sshopts $IP -- grep Trace /var/log/cloud-init.log
+        ssh $sshopts $IP -- systemd-analyze
+        ssh $sshopts $IP -- systemd-analyze blame
+        ssh $sshopts $IP -- cloud-init analyze show
+        ssh $sshopts $IP -- cloud-init analyze blame
         echo 'Networking config'
-        ssh "${sshopts[@]}" $IP -- cat $netcfg
-
-        scp "${sshopts[@]}" setup_proposed.sh $IP:.
-        ssh "${sshopts[@]}" $IP -- sudo bash ./setup_proposed.sh 2>&1 | egrep 'cloud-init'
-        ssh "${sshopts[@]}" $IP -- dpkg-query --show cloud-init
-        ssh "${sshopts[@]}" $IP -- sudo hostname SRU-didnt-work
-        ssh "${sshopts[@]}" $IP -- sudo rm -f $netcfg
-        ssh "${sshopts[@]}" $IP -- sudo cloud-init clean --logs --reboot
+        ssh $sshopts $IP -- cat $netcfg
 
+        scp $sshopts setup_proposed.sh $IP:.
+        ssh $sshopts $IP -- sudo bash ./setup_proposed.sh 2>&1 | egrep 'cloud-init'
+        ssh $sshopts $IP -- dpkg-query --show cloud-init
+        ssh $sshopts $IP -- sudo hostname SRU-didnt-work
+        ssh $sshopts $IP -- sudo rm -f $netcfg
+        if [ "" = "$nics" ]; then
+            ssh $sshopts $IP -- sudo cloud-init clean --logs --reboot
+        else
+	    echo "Shutting down instance and adding IPv6 config to nic"
```

Comment:
Hrm, how many scenarios do we want to verify, we have quite a few, maybe we can combine them all?

1) number of nics
2) ipv4
3) ipv6
4) advanced networking

I think if we create a VM with two nics, enabled Advanced networking, and assigned mutiple v4 and v6 addresses to each interface, that ought to cover all of the paths we need in a single instance.

We need to create the VM with that config, *then* upgrade it rather then upgrading and then applying the config.  
This allows us to verify that before this SRU things like additional IPV6 addrs and other features/fixes are now present after upgrading.

Reply:
Yes I think we can  combine where possible. This script already combines 2-4 in one test. It doesn't currently address multi-nic.  or 5. multi-ip of same type on one interface. We can followup with that in an iteration I believe.

Also, yesterday this script was breaking on Bionic/Disco/Eoan because I was trying to launch dual stack ipv4/ipv6  which broke on our current published cloud-init (because of ipv4 != ipv6 route-metrics). So we couldn't launch the vm out of the gate with the newly supported feature. Generally though, as long as the feature gap isn't catastrophic, we should be able to launch the initial config prior to upgrade and see feature improvement with -proposed.

Reply:
OK, I forgot that we're fixing an issue with dual stack.  so for now, separate.  We do need to sort the multi-nic case out though.